rpcs/blockService.go

```diff
@@ -301,11 +302,34 @@ const datatypeUnsupportedErrMsg = "requested data type is unsupported"
 func (bs *BlockService) handleCatchupReq(ctx context.Context, reqMsg network.IncomingMessage) {
 	target := reqMsg.Sender.(network.UnicastPeer)
 	var respTopics network.Topics
+	var n uint64
 
 	defer func() {
+		if n > 0 {
+			reqMsg.Callback = func() {
+				bs.mu.Lock()
+				bs.memoryUsed -= n
+				bs.mu.Unlock()
+			}
+			bs.mu.Lock()
+			bs.memoryUsed += n
+			bs.mu.Unlock()
+		}
 		target.Respond(ctx, reqMsg, respTopics)
 	}()
 
+	// If we are over-capacity, we will not process the request
+	// respond to sender with error message
+	memUsed := atomic.LoadUint64(&bs.memoryUsed)
+	if memUsed > bs.memoryCap {
```

Comment:
Do we want to just not return a response here? Or at least not give the peer memory information

Reply:
Great question. 
We need to do something similar to the HTTP request. There we got it for free, using the protocol retry message. 
Here, we need to implement on the requester side, how to react to this error. 

In this case, the requester should request it from another peer. It cannot wait and request again, since these requests are time sensitive.

Reply:
Returning an error in that case should still be good enough, but 4 second timeouts do happen somewhat regularly right now